src/openfl/_internal/renderer/canvas/CanvasBitmap.hx

```diff
@@ -29,14 +29,16 @@ class CanvasBitmap {
 			context.globalAlpha = bitmap.__worldAlpha;
 			var transform = bitmap.__renderTransform;
 			var scrollRect = bitmap.__scrollRect;
+			var scale = renderSession.scale;
+			@:privateAccess var sizeScale = scale / bitmap.__bitmapData.__scale;
 			
-			if (renderSession.roundPixels || bitmap.__snapToPixel()) {
+			if (renderSession.roundPixels || bitmap.__snapToPixel ()) {
 				
-				context.setTransform (transform.a, transform.b, transform.c, transform.d, Math.round (transform.tx), Math.round (transform.ty));
+				context.setTransform (transform.a * sizeScale, transform.b, transform.c, transform.d * sizeScale, Math.round (transform.tx * scale), Math.round  (transform.ty * scale));
```

Comment:
maybe at this point we should remove the copy-paste of `context.setTransform` call and just save `tx/ty` in local vars and round them if needed.

same in other places

Reply:
Well not sure, I prefer it this way because the "scaling" for resize and translate is at one places. If we move tx/ty in vars we would need to do the mulitplication there, and another multiplication for resizing in setTransform. This is not too bad as well, but I think is better readable if you see it in one line.